analysis/quantifier-comparisons/11-five-prime-benchmarking.Rmd

```diff
@@ -0,0 +1,372 @@
+  facet_grid(~ sample) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("Mito Percent") + 
+  xlab("")
+```
+```{r}
+ggplot(coldata_qc_common, aes(x = tool, y = sum, fill = tool)) + 
+  geom_boxplot() + 
+  facet_grid(~ sample) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("UMI/cell") + 
+  xlab("") + 
+  ylim(c(0,30000))
+```
+```{r}
+ggplot(coldata_qc_common, aes(x = tool, y = detected, fill = tool)) + 
+  geom_boxplot() + 
+  facet_grid(~ sample) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("Genes detected/cell") + 
+  xlab("") + 
+  ylim(c(0,6000))
+```
+Again, when restricting to shared cells, the distributions are mostly similar although it looks like Alevin-fry might be giving slightly higher counts and genes detected/cell than Cell Ranger. 
+
+### Shared Cells, Minimum Gene Coverage
+
+We can also restrict the comparisons to only include genes that are found in at least 5% of cells to see if low covered genes are affecting the distributions at all. 
+To do that we will need to recalculate the per cell metrics using only genes found in > 5% of cells. 
+
+```{r}
+# combine all tools into one list
+sce_list <- append(fry_sce, cellranger_sce)
+
+# use addPerCellQC with gene detection threshold
+# only genes with detection in > 5% of cells will be included in the QC calculations
+add_qc_threshold <- function(sce, mito = mito_genes, threshold = 5){
+  # remove previous colData first 
+  colData(sce) <- NULL
+  scpcaTools::add_cell_mito_qc(sce, mito = mito_genes, threshold = threshold)
+}
+
+# apply above function to all sces 
+sce_threshold_list <- sce_list %>%
+    purrr::map(add_qc_threshold)
+```
+
+```{r}
+# merge back into a dataframe for plotting 
+coldata_threshold <- sce_threshold_list %>%
+  purrr::map_df(scpcaTools::coldata_to_df, .id = "quant_id") %>%
+  # join with quant info to get sample and tool information 
+  dplyr::left_join(quant_info, by = c("quant_id" = "quant_dir"))
+```
+
+```{r}
+# look for shared cells only 
+cell_counts_threshold <- coldata_threshold %>%  
+  dplyr::count(cell_id, sample)
+
+common_cells_threshold <- cell_counts_threshold %>%
+  dplyr::filter(n == 2) %>%
+  dplyr::pull(cell_id)
+
+qc_common_threshold <- coldata_threshold %>%
+  dplyr::filter(
+    (cell_id %in% common_cells_threshold) 
+  )
+```
+
+```{r}
+ggplot(qc_common_threshold, aes(x = tool, y = sum, fill = tool)) + 
+  geom_boxplot() + 
+  facet_grid(~ sample) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("UMI/cell") + 
+  xlab("") + 
+  ylim(c(0,30000))
+```
+```{r}
+ggplot(qc_common_threshold, aes(x = tool, y = detected, fill = tool)) + 
+  geom_boxplot() + 
+  facet_grid(~ sample) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("Genes detected/cell") + 
+  xlab("") + 
+  ylim(c(0,1000))
```

Comment:
The fact that the genes per cell drops so precipitously here is interesting, but I guess that probably suggests the data isn't particularly saturated?